GitExtensions/Program.cs

```diff
@@ -159,14 +160,22 @@ private static string GetWorkingDir(string[] args)
                 ////   Repositories.RepositoryHistory.AddMostRecentRepository(Module.WorkingDir);
             }
 
-            if (args.Length <= 1 && string.IsNullOrEmpty(workingDir) && AppSettings.StartWithRecentWorkingDir)
+            if (args.Length <= 1 && workingDir == null && AppSettings.StartWithRecentWorkingDir)
             {
                 if (GitModule.IsValidGitWorkingDir(AppSettings.RecentWorkingDir))
                 {
                     workingDir = AppSettings.RecentWorkingDir;
                 }
             }
 
+            if (workingDir == null)
```

Comment:
Tested too: This breaks AppSettings.StartWithRecentWorkingDir (and reverses #4740 as well as "fixes" 4860).
But the AppSettings.StartWithRecentWorkingDir can maybe be added somewhere else.
(Looking at it, rebasing on to of master.)
Maybe the PR should be merged and fixed later?

The following should be changed too:

            if (args.Length <= 1 && workingDir == null && AppSettings.StartWithRecentWorkingDir)
....
            if (workingDir == null)